gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/cluster/BrokerTopologyManagerImpl.java

```diff
@@ -17,59 +17,44 @@
 import io.camunda.zeebe.gateway.Loggers;
 import io.camunda.zeebe.protocol.impl.encoding.BrokerInfo;
 import io.camunda.zeebe.scheduler.Actor;
-import io.camunda.zeebe.scheduler.ActorSchedulingService;
-import io.camunda.zeebe.scheduler.future.ActorFuture;
-import io.camunda.zeebe.scheduler.future.CompletableActorFuture;
-import io.camunda.zeebe.topology.GatewayClusterTopologyService;
+import io.camunda.zeebe.topology.TopologyUpdateNotifier.TopologyUpdateListener;
 import io.camunda.zeebe.topology.state.ClusterTopology;
 import java.util.HashSet;
 import java.util.Set;
-import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
 import java.util.function.Supplier;
 import org.slf4j.Logger;
 
 public final class BrokerTopologyManagerImpl extends Actor
-    implements BrokerTopologyManager,
-        ClusterMembershipEventListener,
-        GatewayClusterTopologyService.Listener {
+    implements BrokerTopologyManager, ClusterMembershipEventListener, TopologyUpdateListener {
 
   private static final Logger LOG = Loggers.GATEWAY_LOGGER;
-
-  private final AtomicReference<BrokerClusterStateImpl> topology;
+  private volatile BrokerClusterStateImpl topology = new BrokerClusterStateImpl();
```

Comment:
:question: Why was this changed from `AtomicReference`? Agree it doesnot make much difference as concurrent threads only read them. But I think the semantics is clear with AtomicReference.

Reply:
IMO the previous approach with `AtomicReference` was not sound anyway because we usually just did a `get`, then made some calculations and then `set` the result. So this already depended on single-threaded execution through the actor to be safe.